File: library/src/main/java/cz/developer/library/adapter/DebugViewAdapter.java
package cz.developer.library.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckBox;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import cz.developer.library.R;
import cz.developer.library.model.DebugViewItem;

/**
 * Created by cz on 1/11/17.
 */

public class DebugViewAdapter extends BaseAdapter{
    private final LayoutInflater layoutInflater;
    private final List<DebugViewItem> viewItems;
    private final List<DebugViewItem> selectItems;

    public DebugViewAdapter(Context context, List<DebugViewItem> viewItems) {
        this.layoutInflater=LayoutInflater.from(context);
        this.selectItems =new ArrayList<>();
        this.viewItems = new ArrayList<>();
        if(null!=viewItems){
            this.viewItems.addAll(viewItems);
            this.selectItems.addAll(viewItems);
        }
    }

    @Override
    public int getCount() {
        return viewItems.size();
    }

    @Override
    public DebugViewItem getItem(int position) {
        return viewItems.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    public List<DebugViewItem> getSelectItems(){
        return selectItems;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(null==convertView){
            convertView=layoutInflater.inflate(R.layout.debug_view_item,parent,false);
        }
        DebugViewItem item = getItem(position);
        TextView viewName= (TextView) convertView.findViewById(R.id.tv_view_name);
        TextView viewDesc= (TextView) convertView.findViewById(R.id.tv_view_desc);
        CheckBox checkBox= (CheckBox) convertView.findViewById(R.id.check_box);
        viewName.setText(item.viewName);
        viewDesc.setText(item.viewInfo);
        checkBox.setChecked(selectItems.contains(item));
        return convertView;
    }

    public void selectItem(int position) {
        DebugViewItem item = getItem(position);
        if(selectItems.contains(item)){
            selectItems.remove(item);
        } else {
            selectItems.add(item);
        }
        notifyDataSetChanged();
    }
}

File: library/src/main/java/cz/developer/library/view/adapter/DebugViewAdapter.kt
package cz.developer.library.ui.view.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.CheckBox
import android.widget.TextView

import java.util.ArrayList

import cz.developer.library.R
import cz.developer.library.ui.view.DebugViewFragment

/**
 * Created by cz on 9/5/2017.
 */
internal class DebugViewAdapter(context: Context, viewItems: List<DebugViewFragment.Companion.Item>?) : BaseAdapter() {
    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private val viewItems: MutableList<DebugViewFragment.Companion.Item>
    private val selectItems: MutableList<DebugViewFragment.Companion.Item>

    init {
        this.selectItems = ArrayList<DebugViewFragment.Companion.Item>()
        this.viewItems = ArrayList<DebugViewFragment.Companion.Item>()
        if (null != viewItems) {
            this.viewItems.addAll(viewItems)
            this.selectItems.addAll(viewItems)
        }
    }

    override fun getCount(): Int =viewItems.size

    override fun getItem(position: Int): DebugViewFragment.Companion.Item =viewItems[position]

    override fun getItemId(position: Int): Long =position.toLong()

    fun getSelectItems(): List<DebugViewFragment.Companion.Item> =selectItems

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        if (null == convertView) {
            convertView = layoutInflater.inflate(R.layout.debug_view_item, parent, false)
        }
        val item = getItem(position)
        val viewName = convertView?.findViewById(R.id.tv_view_name) as TextView
        val viewDesc = convertView?.findViewById(R.id.tv_view_desc) as TextView
        val checkBox = convertView?.findViewById(R.id.check_box) as CheckBox
        viewName.text = item.title
        viewDesc.text = item.desc
        checkBox.isChecked = selectItems.contains(item)
        return convertView
    }

    fun selectItem(position: Int) {
        val item = getItem(position)
        if (selectItems.contains(item)) {
            selectItems.remove(item)
        } else {
            selectItems.add(item)
        }
        notifyDataSetChanged()
    }
}
